### ISSUE TITLE
Test failures in Genus2 curves, elliptic curves, and lattices

### ISSUE THREAD

**Issue by sehlen** [LMFDB/lmfdb#1509]
On the current master, I get the following test failures:

```

======================================================================
ERROR: test_stats (lmfdb.genus2_curves.test_genus2_curves.Genus2Test)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ehlen/lmfdb/lmfdb/genus2_curves/test_genus2_curves.py", line 8, in test_stats
    L = self.tc.get('/Genus2Curve/Q/stats')
  File "/usr/local/sage/sage-1/local/lib/python2.7/site-packages/Werkzeug-0.11.5-py2.7.egg/werkzeug/test.py", line 778, in get
    return self.open(*args, **kw)
  File "/usr/local/sage/sage-1/local/lib/python2.7/site-packages/Flask-0.10.1-py2.7.egg/flask/testing.py", line 108, in open
    follow_redirects=follow_redirects)
  File "/usr/local/sage/sage-1/local/lib/python2.7/site-packages/Werkzeug-0.11.5-py2.7.egg/werkzeug/test.py", line 751, in open
    response = self.run_wsgi_app(environ, buffered=buffered)
  File "/usr/local/sage/sage-1/local/lib/python2.7/site-packages/Werkzeug-0.11.5-py2.7.egg/werkzeug/test.py", line 668, in run_wsgi_app
    rv = run_wsgi_app(self.application, environ, buffered=buffered)
  File "/usr/local/sage/sage-1/local/lib/python2.7/site-packages/Werkzeug-0.11.5-py2.7.egg/werkzeug/test.py", line 871, in run_wsgi_app
    app_rv = app(environ, start_response)
  File "/usr/local/sage/sage-1/local/lib/python2.7/site-packages/Flask-0.10.1-py2.7.egg/flask/app.py", line 1836, in __call__
    return self.wsgi_app(environ, start_response)
  File "/usr/local/sage/sage-1/local/lib/python2.7/site-packages/Flask-0.10.1-py2.7.egg/flask/app.py", line 1820, in wsgi_app
    response = self.make_response(self.handle_exception(e))
  File "/usr/local/sage/sage-1/local/lib/python2.7/site-packages/Flask-0.10.1-py2.7.egg/flask/app.py", line 1403, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/sage/sage-1/local/lib/python2.7/site-packages/Flask-0.10.1-py2.7.egg/flask/app.py", line 1817, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/sage/sage-1/local/lib/python2.7/site-packages/Flask-0.10.1-py2.7.egg/flask/app.py", line 1477, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/sage/sage-1/local/lib/python2.7/site-packages/Flask-0.10.1-py2.7.egg/flask/app.py", line 1381, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/sage/sage-1/local/lib/python2.7/site-packages/Flask-0.10.1-py2.7.egg/flask/app.py", line 1475, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/sage/sage-1/local/lib/python2.7/site-packages/Flask-0.10.1-py2.7.egg/flask/app.py", line 1461, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/ehlen/lmfdb/lmfdb/genus2_curves/genus2_curve.py", line 126, in statistics
    info = { 'counts': g2cstats().counts(), 'stats': g2cstats().stats() }
  File "/home/ehlen/lmfdb/lmfdb/genus2_curves/genus2_curve.py", line 382, in stats
    self.init_g2c_stats()
  File "/home/ehlen/lmfdb/lmfdb/genus2_curves/genus2_curve.py", line 420, in init_g2c_stats
    value_string = attr['format'](value) if 'format' in attr else value
  File "/home/ehlen/lmfdb/lmfdb/genus2_curves/web_g2c.py", line 268, in aut_group_name
    return group_dict[name]
NameError: global name 'group_dict' is not defined

======================================================================
FAIL: test_dl_all (lmfdb.elliptic_curves.test_ell_curves.EllCurveTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ehlen/lmfdb/lmfdb/elliptic_curves/test_ell_curves.py", line 72, in test_dl_all
    assert '[1, -1, 1, -3, 3]' in L.data
AssertionError

======================================================================
FAIL: test_by_curve_label (lmfdb.genus2_curves.test_genus2_curves.Genus2Test)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ehlen/lmfdb/lmfdb/genus2_curves/test_genus2_curves.py", line 21, in test_by_curve_label
    assert 'square of an elliptic curve' in L.data and 'E_6' in L.data
AssertionError

======================================================================
FAIL: test_by_url_curve_label (lmfdb.genus2_curves.test_genus2_curves.Genus2Test)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ehlen/lmfdb/lmfdb/genus2_curves/test_genus2_curves.py", line 57, in test_by_url_curve_label
    assert 'simple over' in L.data and 'G_{3,3}' in L.data
AssertionError

======================================================================
FAIL: test_latticeZ2 (lmfdb.modlmf.test_lattice.HomePageTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ehlen/lmfdb/lmfdb/modlmf/test_lattice.py", line 71, in test_latticeZ2
    assert '0.785398163397448309615660845820' in L #Z2 lattice
AssertionError

----------------------------------------------------------------------
Ran 256 tests in 270.347s

FAILED (SKIP=7, errors=1, failures=4)
```

The genus2 curve failures seem to have been introduced recently. When I pulled most changes seem to have been in that component. The elliptic curve failure seems to be a data problem, no?

**Comment by sehlen** [comment]
PS: I don't really know why the lattice test fails, though. I can see that string on the page.
However, why is this test_lattice.py actually in lmfdb/modlmf/?

**Comment by sehlen** [comment]
The "real" lattice test in lmfdb/lattice passes btw and it has the \dots that are shown on the page, so maybe that's the reason...

**Comment by sanni85** [comment]
I have no clue why is there, it should not be.

**Comment by sanni85** [comment]
Sorry I know why is there: when I was working on the mod l modular forms I took things from lattices and I forgot to remove that file.... I will delete it now.

**Comment by AndrewVSutherland** [comment]
The genus 2 test failures are now fixed on master (they were inadvertently introduced in a recent cleanup), thanks for letting me know (and shame on me for not running my own test script!).

**Comment by JohnCremona** [comment]
@AndrewVSutherland It happens to all os us (see #1508)!

**Comment by AndrewVSutherland** [comment]
A fix for the elliptic_curve test failure was included in PR #1512.  For the current version on master (as of commit https://github.com/LMFDB/lmfdb/commit/aabcfb3ad84ab1ba9fbcd8461aefad7d0bf7f169) all tests pass.  Closing.
